### PR TITLE
[ready] Include datasources in route annotation.

### PR DESCRIPTION
**Pull request by danpat**
This PR adds datasource information to the route annotation.
When using multiple traffic sources, this allows you to see the makeup of discovered route.

### CODE REVIEW

File: include/engine/guidance/assemble_geometry.hpp [hunk 1]
```diff
@@ -79,8 +83,14 @@ inline LegGeometry assembleGeometry(const datafacade::BaseDataFacade &facade,
...
     geometry.segment_distances.push_back(cumulative_distance);
+
+    std::vector<DatasourceID> forward_datasources;
+    facade.GetUncompressedDatasources(target_node.forward_packed_geometry_id, forward_datasources);
+
     geometry.annotations.emplace_back(
-        LegGeometry::Annotation{current_distance, target_node.forward_weight / 10.});
+        LegGeometry::Annotation{current_distance,
+                                target_node.forward_weight / 10.,
+                                forward_datasources[target_node.fwd_segment_position]});
```
Comment (daniel-j-h) [inline include/engine/guidance/assemble_geometry.hpp]: Hm maybe this is unrelated, but here and above the `.emplace_back(T{arg0, arg1, ..})` calls should probably be `.push_back(T{arg0, arg1, ...})` calls --- `push_back` has an rvalue overload.

File: scripts/format.sh [hunk 2]
```diff
@@ -7,8 +7,33 @@
...
+    V=$(clang-format --version)
+    if [[ $V != *3.8* ]] ; then
+        echo "clang-format is not 3.8 (returned ${V})"
+        exit 1
+    fi
+else
+    echo "No appropriate clang-format found (expected clang-format-3.8, or clang-format)"
+    exit 1
+fi
+
```
Comment (daniel-j-h) [inline scripts/format.sh]: :+1:

**Comment by TheMarex** [comment]
Missing here:
- [ ] Changelog entry for extended API
- [ ] Docs about annotation should be extended a little bit. I don't see node-ids mentioned there

**Comment by daniel-j-h** [comment]
i686 build still breaks https://github.com/Project-OSRM/osrm-backend/blob/fd6bb5ec1f48e341218e64d015927b5b9288f94c/features/testbot/matching.feature#L124

**Pull request merged**